private/buf/cmd/buf/command/push/push.go

```diff
@@ -139,6 +150,16 @@ func run(
 	if len(flags.Tags) > 0 && flags.Draft != "" {
 		return appcmd.NewInvalidArgumentErrorf("--%s (-%s) and --%s cannot be used together.", tagFlagName, tagFlagShortName, draftFlagName)
 	}
+	if flags.CreateVisibility != "" {
+		if !flags.Create {
+			return appcmd.NewInvalidArgumentErrorf("Cannot set --%s without --%s.", createVisibilityFlagName, createFlagName)
+		}
```

Comment:
Nit: if both flags are required, and cannot be sent one w/out the other, may be it makes more sense having just a single flag called --create-with-visibility {REPO_VISIBILITY}? It'd save this double check.

Reply:
Good point. Atm I cannot think of another option besides visibility, but you're right, in case of another flag it'd be very weird. And if we endup making private the default, then the `--create-with-visibility` can just become optional and only the `--create` would be required.